### ISSUE TITLE
Add Batch Normalization

### ISSUE THREAD

**Issue by Daniel-Liu-c0deb0t** [Daniel-Liu-c0deb0t/Java-Machine-Learning#4]
It would be great if someone adds batch normalization as a layer. Make sure to include both the forward propagation code and the backpropagation code. Also add comments for the derivative equations if they are not straightforward (for an example, take a look at the GRU layer code). Finally, make sure the trainable variables are able to be saved!
